build.gradle

```diff
@@ -202,6 +202,22 @@ model {
     }
 }
 
+project(':devApp') {
+    apply plugin: 'java'
+    apply plugin: 'application'
+
+    dependencies {
+        compile project(':')
+        runtime files(project(':').nativeTestFilesJar.archivePath)
+    }
+
+    sourceSets.main.java.srcDirs = [ "${rootDir}/src/dev/java" ]
```

Comment:
I wish gradle supported Java as components like cpp, not only subprojects. Its a much cleaner layout imo.